library/report.inc

```diff
@@ -109,23 +109,52 @@ subscriber_employer_state => xl('Subscriber Employer State') . ": ",
 subscriber_employer_country => xl('Subscriber Employer Country') . ": "
 );
 
+$insurance_data_array_custom = array(
+provider_name => xl('Provider') . ": ",
+line1 => xl('Insurance Address') . ": ",
+city => xl('Insurance City') . ": ",
+state => xl('Insurance State') . ": ",
+zip => xl('Insurance Zip') . ": ",
+plan_name => xl('Plan Name') . ": ",
+policy_number => xl('Policy Number') . ": ",
+group_number => xl('Group Number') . ": ",
+subscriber_fname => xl('Subscriber First Name') . ": ",
+subscriber_mname => xl('Subscriber Middle Name') . ": ",
+subscriber_lname => xl('Subscriber Last Name') . ": ",
+subscriber_relationship => xl('Subscriber Relationship') . ": ",
+subscriber_ss => xl('Subscriber SS') . ": ",
+subscriber_DOB => xl('Subscriber Date of Birth') . ": ",
+subscriber_phone => xl('Subscribter Phone') . ": ",
+subscriber_street => xl('Subscriber Address') . ": ",
+subscriber_postal_code => xl('Subscriber Zip') . ": ",
+subscriber_city => xl('Subscriber City') . ": ",
+subscriber_state => xl('Subscriber State') . ": ",
+subscriber_country => xl('Subscriber Country') . ": ",
+subscriber_employer => xl('Subscriber Employer') . ": ",
+subscriber_employer_street => xl('Subscriber Employer Street') . ": ",
+subscriber_employer_city => xl('Subscriber Employer City') . ": ",
+subscriber_employer_postal_code => xl('Subscriber Employer Zip') . ": ",
+subscriber_employer_state => xl('Subscriber Employer State') . ": ",
+subscriber_employer_country => xl('Subscriber Employer Country') . ": "
+);
+
 function getPatientReport($pid)
 {
-	$sql = "select * from patient_data where pid='$pid' order by date ASC";
-	$res = sqlStatement("$sql");
-	while($list = sqlFetchArray($res))
-	{
-		while(list($key, $value) = each($list))
-		{
-			if ($ret[$key]['content'] != $value && $ret[$key]['date'] < $list['date'])
-			{
-				$ret[$key]['title'] = $key;
-				$ret[$key]['content'] = $value;
-				$ret[$key]['date'] = $list['date'];
-			}
-		}
-	}
-	return $ret;
+    $sql = "select * from patient_data where pid='$pid' order by date ASC";
```

Comment:
`$sql = "select * from patient_data where pid='$pid' order by date ASC";`
Change to `$sql = "select * from patient_data where pid = ? order by date ASC;";`
If that doesn't work then...
 `$sql = "select * from patient_data where pid= ".$pid." order by date ASC;";`

Reply:
Ooops.  That assumes a numeric pid...
Make that either the ordinal ? or use '".$pid."' (single, double,dot,var,dot, double, single).

Reply:
That is outside of the scope of this pr. It you look down the file there are a huge number that need to be fixed. I offered to change them in another pr. Tony wants the pr's to be what they are, nothing added or taken away.

Reply:
Otay.